tests-functional/tests/test_backup_mnemonic_and_restore.py

```diff
@@ -0,0 +1,187 @@
+        # Initialize backend client and restore account using user_mnemonic.
+        restored_account = StatusBackend()
+        restored_account.init_status_backend()
+        restored_account.restore_account_and_login(user=user_mnemonic)
+
+        # Request getAccounts and check restoreed accounts attributes.
+        get_accounts_response = restored_account.accounts_service.get_accounts()
+        assert_response_attributes(get_accounts_response.get("result", {}), user_mnemonic.accounts)
+
+        # Request getSettings and check restoreed profile data attributes.
+        get_settings_response = restored_account.settings_service.get_settings()
+        restored_settings = get_settings_response.get("result", {})
+        assert_response_attributes(restored_settings, user_mnemonic.profile_data)
+
+        assert restored_settings.get("mnemonic", None) is None
+        assert restored_settings.get("address", None)
+
+    @pytest.mark.parametrize(
+        "mnemonic_size",
+        [1, 3, 7, 13, 29],
+    )
+    def test_restore_with_arbitrary_size_mnemonics(self, mnemonic_size):
+        # Restore with an arbitrary length mnemonic
+        user = user_mnemonic_12
+        user.passphrase = " ".join("".join(random.choice(string.ascii_lowercase) for _ in range(random.randint(2, 10))) for _ in range(mnemonic_size))
+
+        restored_account = StatusBackend()
+        restored_account.init_status_backend()
+
+        restored_account.restore_account_and_login(user=user)
+        get_settings_response = restored_account.settings_service.get_settings()
+        restored_settings = get_settings_response.get("result", {})
+        assert restored_settings.get("mnemonic", None) is None
+        assert restored_settings.get("address", None)
+
+    def test_restore_with_mnemonic_with_special_chars(self):
+        # Restore with an mnemonic with special chars
+        user = user_mnemonic_12
+        user.passphrase = "<>?`~!@#$%^&*()_+1 $fgdg ^&*()"
+
+        restored_account = StatusBackend()
+        restored_account.init_status_backend()
+
+        restored_account.restore_account_and_login(user=user)
+        get_settings_response = restored_account.settings_service.get_settings()
+        restored_settings = get_settings_response.get("result", {})
+        assert restored_settings.get("mnemonic", None) is None
+        assert restored_settings.get("address", None)
+
+    def test_restore_with_empty_mnemonic(self):
+        # Restore with empty mnemonic isn't allowed
+        user = user_mnemonic_12
+
+        restored_account = StatusBackend()
+        restored_account.init_status_backend()
+        restored_account.restore_account_and_login(user=user)
+        restored_account._set_display_name()
+        data = restored_account._create_account_request(user)
+        data["mnemonic"] = ""
+        restored_account_response = restored_account.api_request("RestoreAccountAndLogin", data)
```

Comment:
I'm not following, you call `RestoreAccountAndLogin` right after calling it just before, on the same `StatusBackend` instance 🤔 

This should not work until you call `Logout`.  Or call it on a different `StatusBackend` instance.

Reply:
Well it makes sense, you can recover the account that's already there. I agree the error could be more meaningful though